frontend/src/screens/internal-apps/Bitrefill.tsx

```diff
@@ -0,0 +1,139 @@
+import { Invoice } from "@getalby/lightning-tools";
+import React, { useEffect } from "react";
+import AppHeader from "src/components/AppHeader";
+import FormattedFiatAmount from "src/components/FormattedFiatAmount";
+import Loading from "src/components/Loading";
+import {
+  AlertDialog,
+  AlertDialogAction,
+  AlertDialogCancel,
+  AlertDialogContent,
+  AlertDialogFooter,
+  AlertDialogHeader,
+  AlertDialogTitle,
+} from "src/components/ui/alert-dialog";
+import { useToast } from "src/components/ui/use-toast";
+import { PayInvoiceResponse } from "src/types";
+import { request } from "src/utils/request";
+
+export function Bitrefill() {
+  const { toast } = useToast();
+  const [open, setOpen] = React.useState(false);
+  const [loading, setLoading] = React.useState(false);
+  const [invoice, setInvoice] = React.useState<Invoice>();
+
+  useEffect(() => {
+    const handleIframeMessage = (event: MessageEvent) => {
+      if (event.origin === "https://embed.bitrefill.com" && event.data) {
+        // Some of the events have objects in event.data 🤷‍♂️
+        try {
+          const parsedData = JSON.parse(event.data);
+
+          if (parsedData.event === "payment_intent") {
+            setOpen(true);
+            const invoice = new Invoice({ pr: parsedData.paymentAddress });
+            setInvoice(invoice);
+          }
+        } catch (e) {
+          /* empty */
+        }
+      }
+    };
+
+    window.addEventListener("message", handleIframeMessage);
+
+    return () => {
+      window.removeEventListener("message", handleIframeMessage);
+    };
+  }, []);
+
+  async function confirmPayment() {
+    if (!invoice) {
+      return;
+    }
+
+    try {
+      setLoading(true);
+      const payInvoiceResponse = await request<PayInvoiceResponse>(
+        `/api/payments/${invoice.paymentRequest}`,
+        {
+          method: "POST",
+          body: JSON.stringify({
+            amount: invoice.satoshi * 1000,
```

Comment:
I don't think the amount should be set, it's for 0-amount invoices. Bitrefill will always have one set

Reply:
Thanks for the hint, didn't pay enough attention.